### PR TITLE
Preview/DEX 211 Apaya

### PR DESCRIPTION
**Pull request by EvansPie**
DEX-211

# What this PR does

Add Apaya flow

# Notable decisions & other stuff

List & explain ...

# Instructions on how to test this

Let other reviewers know how they can test this.

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [x] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/User Interface/Vault/VaultPaymentMethodViewController.swift [hunk 1]
```diff
@@ -250,6 +250,9 @@ extension VaultedPaymentInstrumentsViewController: UITableViewDataSource, UITabl
             // It will reload the payment instrument on the Universal Checkout view.
             delegate?.reload()
         } else {
+
+            let methodId = viewModel.paymentMethods[indexPath.row].token
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/User Interface/Vault/VaultPaymentMethodViewController.swift:254]: Is this used for anything?

File: Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewController.swift [hunk 2]
```diff
@@ -56,9 +47,15 @@ internal class PrimerWebViewController: PrimerViewController, WKNavigationDelega
         webView.navigationDelegate = self // Control which sites can be visited
         view = webView
         if let url = url {
+            let frameworkVersion = Bundle.primerFramework.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String
+            
             var request = URLRequest(url: url)
             request.timeoutInterval = 60
-            request.allHTTPHeaderFields = headerFields
+            request.allHTTPHeaderFields = [
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewController.swift:54]: I feel that this logic + `frameworkVersion` could be extracted in another function

File: Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewModel.swift [hunk 3]
```diff
@@ -9,143 +9,72 @@
...
+            loadClientToken { [weak self] result in
                 switch result {
                 case .failure(let error):
-                    self?.navigate(.failure(error))
+                    self?.onCompletion?(.failure(error))
                 case .success:
-                    guard let strongSelf = self, strongSelf.configDidLoad() else {
-                        self?.navigate(.failure(PrimerError.configFetchFailed))
+                    guard let self = self, self.configDidLoad() else {
+                        self?.onCompletion?(.failure(PrimerError.configFetchFailed))
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewModel.swift:52]: This doesn't work on XCode 12.4
I got a "Variable declared in 'guard' condition is not usable in its body'

File: Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewController.swift (hunk 2 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewController.swift]: Done!

File: Sources/PrimerSDK/Classes/User Interface/Vault/VaultPaymentMethodViewController.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/Vault/VaultPaymentMethodViewController.swift]: Nope, fixed!

File: Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewModel.swift (hunk 3 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewModel.swift:52]: Oh that's perfectly fine on Xcode > 12.5. I reverted it.

**Pull request merged**